app/uk/gov/hmrc/gform/submission/handlebars/HandlebarsHttpApiSubmitter.scala

```diff
@@ -52,30 +73,62 @@ class RealHandlebarsHttpApiSubmitter[F[_]](
             FocussedHandlebarsModelTree(modelTree),
             TemplateType.Plain
           )
+
         destination.method match {
           case HttpMethod.GET => httpClient.get(uri)
           case HttpMethod.POST =>
-            val body = destination.payload.fold("") {
-              handlebarsTemplateProcessor(
-                _,
-                accumulatedModel,
-                FocussedHandlebarsModelTree(modelTree),
-                destination.payloadType
-              )
-            }
-            httpClient.post(uri, body)
+            val bodies = destination.payload.fold(List(""))(body => parseAndProcessAsList(body))
+            callUntilError(
+              bodies.map((uri, _)),
+              httpClient.post,
+              modelTree.value.formId,
+              destination.id
+            )
           case HttpMethod.PUT =>
-            val body = destination.payload.fold("") {
-              handlebarsTemplateProcessor(
-                _,
-                accumulatedModel,
-                FocussedHandlebarsModelTree(modelTree),
-                destination.payloadType
-              )
-            }
-            httpClient.put(uri, body)
+            val bodies = destination.payload.fold(List(""))(body => parseAndProcessAsList(body))
+            callUntilError(
+              bodies.map((uri, _)),
+              httpClient.put,
+              modelTree.value.formId,
+              destination.id
+            )
```

Comment:
Can we keep handling of `destination.multiRequestPayload` handling login separated from current single payload implementation? This implementation is hard to follow.

Simple `if .. else` should do the trick
```
if (destination.multiRequestPayload) {
   // call parseAndProcessAsList with the assumption it is only for multiRequestPayload
   // Feel free to put code for multiRequestPayload into its own class etc.
} else {

  destination.method match {
    case HttpMethod.GET => httpClient.get(uri)
    case HttpMethod.POST => ... current code unchanged
}
```

Reply:
refactored to keep the current code unchanged when multiRequestPayload = false